eclipse/che-theia
# Clone Git projects with SSH URI

### What does this PR do?
Allows the workspace plugin to handle events when cloning Git projects by SSH uri
When the devfile contain GIT SSH project, the workspace plugin
- performs the secure login to Git repository
- in case of GitHub repository, proposes the user to upload public key
- proposes the user to add auto-generated key to the Git repository by hands
- allows the user to use its custom certificate

### Screenshot/screencast of this PR


![Screenshot from 2021-02-22 14-19-56](https://user-images.githubusercontent.com/1655894/108707869-5247c700-7519-11eb-88c6-87659f298b90.png)

![Screenshot from 2021-02-22 14-20-04](https://user-images.githubusercontent.com/1655894/108707879-5542b780-7519-11eb-9a17-1ec5c19f8016.png)

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/13494

### HappyPath tests related on SSH
- https://github.com/eclipse/che/blob/master/tests/e2e/tests/e2e/GitSsh.spec.ts

### How to test this PR?


Devfile to test
```
---
apiVersion: 1.0.0
metadata:
  generateName: clone-ssh-
projects:
  -
    name: che-theia
    source:
      type: git
      location: "git@github.com:eclipse/che-theia.git"
  -
    name: atlascode
    source:
      type: git
      location: "git@bitbucket.org:atlassianlabs/atlascode.git"
components:
  - type: cheEditor
    alias: che-theia
    reference: https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-995/che_theia_meta.yaml
    memoryLimit: 2G
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
